sdk/inc/azure/iot/az_iot_hub_client.h

```diff
@@ -584,6 +684,135 @@ AZ_NODISCARD az_result az_iot_hub_client_twin_patch_get_publish_topic(
     size_t mqtt_topic_size,
     size_t* out_mqtt_topic_length);
 
+/*
+ *
+ * Properties APIs
+ *
+ */
+
+/**
+ * @brief The MQTT topic filter to subscribe to properties operation responses.
+ * @note Twin MQTT Publish messages will have QoS At most once (0).
+ */
+#define AZ_IOT_HUB_CLIENT_PROPERTIES_RESPONSE_SUBSCRIBE_TOPIC \
+  AZ_IOT_HUB_CLIENT_TWIN_RESPONSE_SUBSCRIBE_TOPIC
+
+/**
+ * @brief Gets the MQTT topic filter to subscribe to desired properties changes.
+ * @note Property MQTT Publish messages will have QoS At most once (0).
+ */
+#define AZ_IOT_HUB_CLIENT_PROPERTIES_PATCH_SUBSCRIBE_TOPIC \
+  AZ_IOT_HUB_CLIENT_TWIN_PATCH_SUBSCRIBE_TOPIC
+
+/**
+ * @brief Properties response type.
+ *
+ */
+typedef enum
+{
+  AZ_IOT_HUB_CLIENT_PROPERTIES_RESPONSE_TYPE_GET
+  = AZ_IOT_HUB_CLIENT_TWIN_RESPONSE_TYPE_GET, /**< A response from a properties "GET" request. */
+  AZ_IOT_HUB_CLIENT_PROPERTIES_RESPONSE_TYPE_DESIRED_PROPERTIES
+  = AZ_IOT_HUB_CLIENT_TWIN_RESPONSE_TYPE_DESIRED_PROPERTIES, /**< A "PATCH" response with a payload
+                                                                containing desired properties. */
+  AZ_IOT_HUB_CLIENT_PROPERTIES_RESPONSE_TYPE_REPORTED_PROPERTIES
+  = AZ_IOT_HUB_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES, /**< A response with the result of the
+                                                                 earlier reported properties. */
+} az_iot_hub_client_properties_response_type;
+
+/**
+ * @brief Properties response.
+ *
+ */
+typedef struct
+{
+  az_iot_hub_client_properties_response_type response_type; /**< Properties response type. */
+  az_iot_status status; /**< The operation status. */
+  az_span
+      request_id; /**< Request ID matches the ID specified when issuing a Get or Patch command. */
+} az_iot_hub_client_properties_response;
```

Comment:
>az_iot_hub_client_properties_response [](http://example.com/codeflow?start=2&length=37)

is version missing here?

These all twin api's are rename, can we do #define for renaming? If we see drift between them, then we can define new api.

Reply:
Part1 - Yes, the version is missing by design but we can talk out.  The issue is that this version from the MQTT topic doesn't help the PnP author at all as near as I can tell (since what you get depends on topic response type).  And it's confusing with the $version that we parse out of desired which is what's really interesting.

Thoughts?  Once I removed this version, it strongly implies not using a #define for this one at least.

Part2 - RE #define or no on other places, I'm going back & forth.  On C SDK it's a bad idea to do this because of ABI concerns (we can't swap to a .c later once we GA) but embedded we have more flexibility since we don't have ABI compat (do we?).  

My motivation for a .c is that if I'm single stepping then it's easier to see what's happening in a debugger than the call "just switching", although that's not insurmountable.

There's also consistency since stuff like parsing PnP commantds "." out needs to be .c, as does this one (I think) so I'm tending to not mix & match.

Reply:
For version: In case of desired property PATCH coming from iothub, then we can use the version in topic, which can reduce one scan on document. But I guess keeping is simple  and getting always from one source is right thing, less confusing for us and customer.

I agree with keeping it function help us more then saving some bytes used in epilogue and prologue, which compiler might optimize if seeing call to other function immediately.